packages/vechain/src/ConnexProviderWrapper.ts

```diff
@@ -24,19 +24,31 @@ export default class ConnexProviderWrapper extends EventEmitter implements Ether
         this._provider
             .then(provider => {
                 provider.emit = this.emit.bind(this);
+            })
+            .catch(error => {
+                this._log(`Error while creating provider: ${error.message}`);
+                throw error;
             });
     }
 
     public setProvider(provider: Promise<Provider>) {
         this._provider.then(provider => {
             provider.emit = provider.emit.bind(provider);
+        })
+        .catch(error => {
+            this._log(`Error while setting provider: ${error.message}`);
```

Comment:
Should the log level be `error` (or equivalent) instead? I think `error.stack` would show more info too.

Reply:
This is the debugger logger, so it doesn't have levels, but will include error stack in next PR

Reply:
Ok, thanks for clarifying!